src/cli.js

```diff
@@ -35,6 +36,21 @@ export default function CLI(argv, cwd) {
     .usage("Usage: $0 <command> [options]")
     .options(globalOptions).group(globalKeys, "Global Options:")
     .commandDir("../lib/commands")
+    .check(function (argv) {
```

Comment:
I would rather we use a [default command](https://github.com/yargs/yargs/blob/master/docs/advanced.md#commands) here than brittle filesystem parsing. It would greatly simplify the logic, too.

```js
    .command("*", "", {}, (argv) => {
      // a default command with no description catches typos or missing subcommands
      log.error("lerna", `${argv._.length ? "Invalid" : "Missing"} command!`);
      log.error("lerna", "Pass --help to see all available commands and options.");

      // exit non-zero instead of throw an error so the CLI can be usefully chained
      process.exitCode = 1;
    })
```

Reply:
💯 much better, thanks